### ISSUE TITLE
use_build_context_synchronously provides incorrect warning in some case

### ISSUE THREAD

**Issue by imrhk** [dart-lang/linter#4223]
`context.mounted` if used with any operator reports warning of `use_build_context_synchronously` but without any operator, it works as expected.

For example,
```dart
 _someMethod(BuildContext context) async {
    await Future.delayed(const Duration(seconds: 8));

    if (context.mounted) {
      Navigator.of(context);
    }
  }
```
does not report any warning but
```dart
_someMethod(BuildContext context) async {
    await Future.delayed(const Duration(seconds: 8));

    if (!context.mounted) return; // or  if (context.mounted == false) or if (context.mounted == true)

    Navigator.of(context);
  }
```
reports warning.

**Comment by bwilkerson** [comment]
The rule does not currently do any kind of flow / data analysis, and it's probably outside the scope of the rule to add such an analysis.

We could consider special casing some common patterns, but I don't think we can / should special case every possible pattern. I don't know whether `if (!context.mounted) return;` is common enough to justify special casing it.

**Comment by imrhk** [comment]
From https://dart-lang.github.io/linter/lints/use_build_context_synchronously.html

It is "Good" to use `if (!context.mounted) return;`. Either the example should be removed or warning should not be shown for this particular case as it is contradictory.

**Comment by kj415j45** [comment]
duplicated with #4007 ?

**Comment by imrhk** [comment]
Closing this as it is duplicate of #4007

**Comment by imrhk** [comment]
Closing this as it is duplicate of https://github.com/dart-lang/linter/issues/4007
